ckan/lib/dictization/model_dictize.py

```diff
@@ -617,6 +616,10 @@ def user_dictize(
             result_dict['plugin_extras'] = copy.deepcopy(
                 plugin_extras) if plugin_extras else plugin_extras
 
+    # check and update custom user data
+    plugin_data = result_dict.pop('plugin_data', None)
+    result_dict['plugin_data'] = copy.deepcopy(
+        plugin_data) if plugin_data else plugin_data
```

Comment:
if plugin_data is returned in the result_dict then all values will be available through the API without validation controlling what is entered or returned. Instead of `include_plugin_extras`, a `plugin_data` dict parameter could be passed that `user_dictize` could fill, and that dict could be included in the context of validators so that store_plugin_data and load_plugin_data would have access to them.

Reply:
sorry im not sure what you mean here, can we discuss on the next meeting or you could give me more details in the comments